Rakesh zrp

rnori-harv opened the pull request:
This commit implements the ZRP API (mostly contained in app.py). The parameters from the model are the same as those specified in the presentation (first_name, last_name, precinct_split, etc.), and the model returns probabilities for each race in the same format as Surgeo. We can add future improvements such as simplifying the input parameters (to ideally just name and address) as well as caching the pickled object and warming it up prior to the query (as Jay suggested) to prevent additional overhead.

rakesh-nori committed:
no large picklefiles here

rakesh-nori committed:
attribtue error bug

rakesh-nori committed:
J fixed attribute errorr bug - zrp endpt works

rakesh-nori committed:
Finished ZRP API

AngelikaAntsmae reviewed (approved):
Good job Rakesh! This is a lot of work and I think we should schedule a call to see if the NC model set-up needs to be changed in order for consistency One thing that I noticed was that the zrp_fe.obj shows up as an empty file on Git. Not sure if it's because of filetype issues or if it actually is an unnecessary file. Maybe you could check it out!

chrisong88 commented on src/app.py:
```diff
@@ -4,6 +4,22 @@
 import pandas as pd 
 import surgeo
+import pickle
+#import zrp_predict
```
Can you please take out this comment on line 8?

chrisong88 commented on src/app.py:
```diff
@@ -38,13 +54,34 @@ def zrp():
+    house_district = request.args.get('house_district')
+    birth_date = request.args.get('birth_date')
+
+    data = {'Name_First': [str(first_name)], 'Name_Last': [str(last_name)], 'Name_Middle': [str(middle_name)], 'Zipcode': [int(zipcode)], 'Precinct_Split': [str(precinct_split)], 'Gender': [str(gender)], 'County_Code': [str(county_code)], 'Congressional_District':[float(congressional_district)], 'Senate_District': [float(senate_district)], 'House_District': [float(house_district)], 'Birth_Date':  ...
```
Style: could you split line 77 into several lines? Right now, it's a little difficult to read all on one line. Something like:

![image](https://user-images.githubusercontent.com/29349539/112421822-0f7e3800-8ced-11eb-8705-3af553e98533.png)

chrisong88 replied:
Even though the statement takes up multiple lines, because it's all within one set of {}, Python doesn't throw an error.

chrisong88 commented on src/app.py:
```diff
@@ -38,13 +54,34 @@ def zrp():
+    preds = zrp_predict.generatePredictions(sample)
+
+
+    preds_data = {'American Indian': str(preds[0][0]), 'Asian Pacific Islander': str(preds[0][1]), 'Black': str(preds[0][2]), 'Hispanic': str(preds[0][3]), 'White': str(preds[0][4]), 'Other': str(preds[0][5]), 'Multi': str(preds[0][6])}
```
Same here: please split up into multiple lines

rnori-harv replied:
@chrisong88 just addressed your comments in the latest commit!

chrisong88 commented on src/zrp_predict.py:
```diff
@@ -0,0 +1,137 @@
+        return(data_fl_co_all_sample_clean)
+
+
+class zrp_feature_engineering():
```
This class is almost identical to ZRPFeatureEngineering class in src/zrp_feature_engineering.py . Is that intentional or is one of them redundant? If so, could you remove the redundant one?

chrisong88 reviewed (changes requested):
Overall, great stuff! Solid work connecting all of the different technical challenges that needed to be addressed.

There's one critical part that I'd like you to address before merging. See the comment at zrp_predict.py.

Also, could you update the README.md documentation with some instructions on which .obj files need to be loaded from the Sharefile and which directories they should be uploaded to when running the app?

rnori-harv committed:
Update app.py

rnori-harv committed:
Delete zrp_feature_engineering.py

rakesh-nori committed:
addressed Chris's comments

rakesh-nori committed:
Addressed Chris's comments

rakesh-nori committed:
edited readme

rnori-harv commented:
Hey @chrisong88! I removed the ZRPFeatureEngineering class from zrp_predict.py in more recent commits so there isn't overlap between the two files, and I just updated the README as well!

chrisong88 commented on src/app.py:
```diff
@@ -74,12 +74,30 @@ def zrp():
+                  'Hispanic': str(preds[0][3]), 
+                  'White': str(preds[0][4]), 
+                  'Other': str(preds[0][5]), 
+                  'Multi': str(preds[0][6])}
```
Perfect!

chrisong88 reviewed (approved):
Nice fixes!

chrisong88 committed:
Merge branch 'dev' into rakesh-zrp
